src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -169,12 +170,13 @@ public boolean isCreated() {
         }
 
         @Override
-        public void put(String modelId, KNNEngine knnEngine, byte[] modelBlob, ActionListener<IndexResponse> listener)
-                throws IOException {
-            String base64Model = Base64.getEncoder().encodeToString(modelBlob);
+        public void put(String modelId, Model model, ActionListener<IndexResponse> listener) throws IOException {
+            String base64Model = Base64.getEncoder().encodeToString(model.getModelBlob());
```

Comment:
Question: Will this help if we directly store base64 version of model in the system index? We can avoid these conversions during indexing.

Reply:
In order to index into `binary` field type, we need to convert to Base64 encoded string.